cli/src/pcluster/aws/ec2.py

```diff
@@ -221,24 +222,62 @@ def get_official_image_id(self, os, architecture, filters=None):
             raise AWSClientError(function_name="describe_images", message="Cannot find official ParallelCluster AMI")
         return max(images, key=lambda image: image["CreationDate"]).get("ImageId")
 
+    def get_official_images(self, version=None, os=None, architecture=None):
+        """Get the list of official images, optionally filtered by os and architecture."""
+        try:
+            owners = ["amazon"]
+            version = version or "*"
+            os = Ec2Client._get_os_suffix(os)
+            architecture = architecture or "*"
+            name = "aws-parallelcluster-{}-{}-{}*".format(version, os, architecture)
+            filters = [{"Name": "name", "Values": [name]}]
+            return self.describe_images(ami_ids=[], owners=owners, filters=filters)
+        except ImageNotFoundError:
+            return []
+
     @AWSExceptionHandler.handle_client_exception
     @Cache.cached
     def get_eip_allocation_id(self, eip):
         """Retrieve the allocation id of an Elastic IP."""
         return self._client.describe_addresses(PublicIps=[eip])["Addresses"][0]["AllocationId"]
 
     @staticmethod
-    def _get_official_image_name_prefix(os, architecture):
-        """Return the prefix of the current official image, for the provided os-architecture combination."""
+    def _get_os_suffix(os):
+        """Return the os suffix to use in an image name prefix."""
         suffixes = {
             "alinux2": "amzn2-hvm",
             "centos7": "centos7-hvm",
             "centos8": "centos8-hvm",
             "ubuntu1804": "ubuntu-1804-lts-hvm",
             "ubuntu2004": "ubuntu-2004-lts-hvm",
         }
+        return suffixes.get(os, "*")
```

Comment:
Not sure this is what we want. We might consider throwing an exception here, which in the `get_official_images` we might convert into a `BadRequestException` (a subclass of `AWSClientError`?)

Reply:
I have decided to return `''` as a suffix if it is not present in the map. This way we return an empty list, which is coherent with what happens in case we specify an architecture or version that does not exist (i.e. we build a name prefix that does not match against any official image name, thus return an empty list).